front/pages/api/v1/w/[wId]/vaults/[vId]/data_sources/[dsId]/search.ts

```diff
@@ -194,7 +195,48 @@ async function handler(
 
   switch (req.method) {
     case "GET": {
-      return handleDataSourceSearch({ req, res, dataSource });
+      // I could not find a way to make the query params be an array if there is only one tag.
+      if (req.query.tags_in && typeof req.query.tags_in === "string") {
+        req.query.tags_in = [req.query.tags_in];
+      }
+      if (req.query.tags_not && typeof req.query.tags_not === "string") {
+        req.query.tags_not = [req.query.tags_not];
+      }
+
+      const r = await DataSourceSearchQuerySchema.safeParse(req.query);
+
+      if (r.error) {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `Invalid request body: ${r.error.message}`,
+          },
+        });
+      }
+      const searchQuery = r.data;
+      const s = await handleDataSourceSearch({ searchQuery, dataSource });
+      if (s.isErr()) {
+        if (s.error.code === "data_source_error") {
+          return apiError(req, res, {
+            status_code: 400,
+            api_error: {
+              type: "data_source_error",
+              message: s.error.message,
+            },
+          });
+        } else {
+          return apiError(req, res, {
+            status_code: 500,
+            api_error: {
+              type: "internal_server_error",
+              message: s.error.message,
+            },
+          });
+        }
+      } else {
+        return res.json(s.value);
```

Comment:
Still some left 😉

Reply:
9 in the whole codebase. Let's do it in another PR.

Reply:
Fine by me, as long as you track it 😉